### PR TITLE
sindri-zkml-tutorial

### PR DESCRIPTION
**Pull request by stasiacarson**
The Sindri zkML tutorial is a walkthrough of how to generate verifiable inference using a small DNN and Sindri while operating a rollup on Rollkit. 

Please review the documentation and confirm the following:
- [x] wording and naming conventions are in line with Rollkit terminology
- [ ] you are able to run a Polaris EVM
- [ ] you are able to generate a smart contract using the SDK
- [ ] you are able to deploy the smart contract on Rollkit
- [ ] you are able to submit a Verify request on Rollkit
- [ ] grammatical and structure checks

## Summary by CodeRabbit

- **New Features**
	- Enhanced the sidebar navigation with a "zkML rollup" tutorial, providing a detailed guide on integrating verifiable machine learning inference into a rollup using zero-knowledge proofs with Sindri.

### CODE REVIEW

**Comment by MSevey** [comment]
@stasiacarson please resolve the conflicts and update the branch so we can see the preview 🙏

**Comment by MSevey** [comment]
@coderabbitai pause

**Comment by MSevey** [comment]
@coderabbitai resolve

**Review by jcstein — CHANGES_REQUESTED** [review]
Let's also add this to overview page! https://rollkit.dev/tutorials/overview

File: tutorials/zkml.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+The rollup will rely on a local Celestia devnet for consensus and ensuring data availability.
+
+We will focus on using a ZK circuit that incorporates a compact deep neural network model to enable verifiable ML inference.
+In this guide, we will deploy the circuit on Sindri, obtain a smart contract verifier (automatically generated by Sindri) for that circuit, deploy it on Rollkit, create a proof of ML inference on Sindri, and verify the proof on-chain.
+
+This approach not only secures the verification process of machine learning models, but also leverages the decentralized security and scalability of Celestia's architecture.
+
+## Prerequisites
+
+- An operational Polaris EVM using Rollkit (see next step)
```
Comment (Manav-Aggarwal) [inline tutorials/zkml.md:15]: Let's link to the polaris EVM tutorial here. It currently has a disclaimer so would be good to link!

**Review by Manav-Aggarwal — CHANGES_REQUESTED** [review]

**Review by Manav-Aggarwal — APPROVED** [review]

File: learn/intro.md [hunk 2]
```diff
@@ -49,6 +49,7 @@ If you're familiar with Rollkit, you may want to skip to the [tutorials section]
 
 - [Wordle app](/tutorials/wordle)
 - [CosmWasm rollup](/tutorials/cosmwasm)
+- [zkML rollup](/tutorials/zkml.md)
```
Comment (jcstein) [inline learn/intro.md:52]: @Manav-Aggarwal I think we should link to the tutorials page on intro.md, but maybe not duplicate everything that is also in the tutorials/overview.md page .what do you think? in short, I think we only need to list them out once in the site, for maintenance reasons (as this page intro.md is outdated and doesn't contain polaris, etc.)

**Comment by jcstein** [comment]
gm @stasiacarson have you done this list successfully?

> Please review the documentation and confirm the following:
> 
> - [x] wording and naming conventions are in line with Rollkit terminology
> - [ ] you are able to run a Polaris EVM
> - [ ] you are able to generate a smart contract using the SDK
> - [ ] you are able to deploy the smart contract on Rollkit
> - [ ] you are able to submit a Verify request on Rollkit
> - [ ] grammatical and structure checks

**Comment by jcstein** [comment]
> @stasiacarson please resolve the conflicts and update the branch so we can see the preview 🙏

@MSevey I think we need to change workflows to allow previews to manually be built for external PRs. if they are automatically built for external PRs, I imagine we could walk ourselves into a situation where spam PRs are eating up many github resources

File: learn/intro.md (hunk 2 above)
Comment (Manav-Aggarwal) [reply learn/intro.md:52]: Yes agreed, made a PR here: https://github.com/rollkit/docs/pull/337

**Comment by MSevey** [comment]
> > > @stasiacarson please resolve the conflicts and update the branch so we can see the preview 🙏
> > 
> > 
> > @MSevey I think we need to change workflows to allow previews to manually be built for external PRs. if they are automatically built for external PRs, I imagine we could walk ourselves into a situation where spam PRs are eating up many github resources
> 
> @jcstein I'll take this as a follow up, but I'm not optimistic that we can do this.
> 
> Currently external (forks) PRs fail due to the security model of forks not having write access, so they fail to push to the preview repo and fail to generate the comment on the PR. This we don't want to change for a number of reasons, including the one you mentioned. But I haven't tested manually triggering an action on a fork.
> 
> I know that you can't do that from the actions tab, as only branches native to this repo are shown.

I couldn't find an obvious way to manually trigger the preview on forks. 
We would need to us the trigger `pull_request_target` to enable forks but it isn't recommended for actions that build code. 
We currently only use the `pull_request_target` trigger on non-code actions, like labels, comments, etc.

My recommendation would be that we just build forks locally to preview.

**Review by jcstein — APPROVED** [review]
utACK

**Comment by jcstein** [comment]
manually building sounds good to me @MSevey. thanks for investigating!

**Pull request merged**